source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,187 @@
+
+## What do you need?
+
+<!-- todo links -->
+
+- Basic knowledge of the Nix language.
+- A working installation of Nix Package Manager or NixOS.
+- nixpkgs path set in NIX_PATH 
+
+## What is a NixOS configuration
+
+A NixOS configuration is a declarative definition of a Linux installation that specifies which packages to install and how to configure them.
+It is written in the Nix language and uses the Nixpkgs repository, its library, packages and modules.
+<!-- [in my dream such a blob would appear as mediawiki style page-preview at mentions of Nix language]: The Nix expression language is a pure, lazy, functional language. Its main purpose is to describe packages, compositions of packages, and variability within packages.[^nixlang] -->
+<!-- todo links: ancors to introductions to library packages and modules are currently not in the manuals -->
+A NixOS configuration is a file that contains a function that is structured acording to the module<!-- todo link -->  convention.
+A NixOS configuration is a function that follows the convention of a module<!-- todo link to reference-->. <!-- remove one version -->
+
+<!-- [old version 2022.08.10 should be deleted] -->
+<!-- The Git repository Nixpkgs provides the necessary infrastructure and information needed to specify a NixOS configuration[^nixpkgs]. -->
+<!-- Nixpkgs contains a library, packages, and modules. -->
+<!-- The library provides the functional infrastructure to conveniently create packages. -->
+<!-- The packages defined in Nixpkgs provide build scripts needed to build software. -->
+<!-- The NixOS modules are a way to configure the system. [previous formulation] provide a configuration infrastructure to generate the system's configuration files. -->
+
+## Starting from default NixOS Configuration
+
+On NixOS, the configuration file is normally located at `/etc/nixos/configuration.nix`.
+A different location can be specified using the environment variable `$NIX_PATH`.
+For CLI commands that use a configuration, the file path can be specified.
+In this guide this strategy is preferred because the goal is not to install or update NixOS, but to debug a configuration.
+
+On NixOS, you can use the `nixos-generate-config` command to create a configuration file that contains some useful defaults and configuration suggestions [^nixosconf].
+You can create a NixOS configuration in your working directory:
+```shell-session
+nixos-generate-config --dir ./
+```
+
+In the working directory you will find two files: `configuration.nix` and `hardware-configuration.nix`.
+The `hardware-configuration.nix` file is specific to your current installation.
+This file is irrelevant for this tutorial because it has no effect inside a virtual machine.
+The `configuration.nix` file contains various suggestions useful for the initial setup of a desktop computer.
+Without the comments the configuration file contains the following content:
```

Comment:
Installation is a bit problematic. Installation of what?
```suggestion
In the working directory you will find two files: `configuration.nix` and `hardware-configuration.nix`.

The `hardware-configuration.nix` file is specific to the hardware the `nix-generate-config` is being run on.
We can ignore that file for this tutorial because it has no effect inside a virtual machine.

The `configuration.nix` file contains various suggestions for the initial setup of a desktop computer.
Without comments the configuration file contains the following content:
```

The `hardware-configuration.nix` file is specific to your current installation.

```